opencti-platform/opencti-front/src/private/components/observations/indicators/DecayChart.tsx

```diff
@@ -0,0 +1,207 @@
+import React, { FunctionComponent } from 'react';
+import { IndicatorDetails_indicator$data } from '@components/observations/indicators/__generated__/IndicatorDetails_indicator.graphql';
+import Chart from '@components/common/charts/Chart';
+import { ApexOptions } from 'apexcharts';
+import moment from 'moment';
+import { useTheme } from '@mui/styles';
+import { Theme } from '@mui/material/styles/createTheme';
+
+interface DecayChartProps {
+  indicator: IndicatorDetails_indicator$data,
+}
+
+const DecayChart : FunctionComponent<DecayChartProps> = ({ indicator }) => {
+  const theme = useTheme<Theme>();
+
+  const decayCurveColor = theme.palette.primary.main;
+  const reactionPointColor = theme.palette.text.primary;
+  const scoreColor = theme.palette.success.main;
+  const revokeColor = theme.palette.secondary.main;
+
+  const chartLabelBackgroundColor = theme.palette.background.paper;
+  const chartInfoTextColor = theme.palette.text.primary;
+  const chartBackgroundColor = theme.palette.background.default;
+  const graphLineThickness = 4;
+
+  // Time in millisecond cannot be set as number in GraphQL because it's too long
+  // So the time in data is stored as Date and must be converted to time in ms to be drawn on the chart.
+  const convertTimeForChart = (time: Date) => {
+    return moment(time).valueOf();
+  };
+
+  // This is the chart serie data, aka the curve.
+  const decayCurveDataPoints: { x: number; y: number }[] = [];
+  if (indicator.decayChartData && indicator.decayChartData.live_score_serie) {
+    indicator.decayChartData.live_score_serie.forEach((dataPoint) => {
+      decayCurveDataPoints.push({
+        x: convertTimeForChart(dataPoint.time),
+        y: dataPoint.score,
+      });
+    });
+  }
+
+  const graphLinesAnnotations = [];
+  // Horizontal lines that shows reaction points
+  if (indicator.decay_applied_rule?.decay_points) {
+    indicator.decay_applied_rule.decay_points.forEach((reactionPoint) => {
+      const lineReactionValue = {
+        y: reactionPoint,
+        borderColor: reactionPoint === indicator.x_opencti_score ? scoreColor : reactionPointColor,
+        label: {
+          borderColor: reactionPoint === indicator.x_opencti_score ? scoreColor : reactionPointColor,
+          offsetY: 0,
+          style: {
+            color: reactionPoint === indicator.x_opencti_score ? scoreColor : chartInfoTextColor,
+            background: chartLabelBackgroundColor,
+          },
+          text: `${reactionPoint}`,
+        },
+      };
+      graphLinesAnnotations.push(lineReactionValue);
+    });
+
+    // Horizontal "red" area that show the revoke zone
+    const revokeScoreArea = {
+      y: indicator.decay_applied_rule.decay_revoke_score + 1, // trick to have a red line even if revoke score is 0
+      y2: 0,
+      borderColor: revokeColor,
+      fillColor: revokeColor,
+      label: {
+        text: `Revoke score: ${indicator.decay_applied_rule.decay_revoke_score}`,
```

Comment:
Should we not translate this?

Reply:
It will be done on the last chunck (after PO reviews on displayed texts)